server/operations/src/main/java/org/kaaproject/kaa/server/operations/service/DefaultOperationsService.java

```diff
@@ -110,160 +111,175 @@
+            LOG.trace("[{}][{}] procesing event sync request {}.", context.getEndpointKey(), context.getRequestHash(), request);
+            EventServerSync eventSyncResponse = processEventSyncResponse(context.getEndpointKey(), context.getRequestHash(),
+                    metaData.getApplicationToken(), request, context.getEndpointProfile());
+            context.setEventSyncResponse(eventSyncResponse);
         }
+        return context;
+    }
 
-        boolean updateProfileRequired = false;
-
-        if (request.getConfigurationSync() != null) {
-            ConfigurationClientSync confSyncRequest = request.getConfigurationSync();
-            LOG.trace("[{}][{}] procesing configuration sync request {}.", endpointId, requestHash, confSyncRequest);
-            LOG.debug("[{}][{}] fetching history for seq numbers {}-{}", endpointId, requestHash, confSyncRequest.getAppStateSeqNumber(), curAppSeqNumber);
-            int startSeqNumber = Math.min(confSyncRequest.getAppStateSeqNumber(), profile.getCfSequenceNumber());
-            LOG.debug("[{}][{}] calculating configuration delta using seq number {}", endpointId, requestHash, startSeqNumber);
-            confSyncRequest.setAppStateSeqNumber(startSeqNumber);
-            HistoryDelta historyDelta = fetchHistory(endpointId, requestHash, metaData.getApplicationToken(), profile, HistorySubject.CONFIGURATION,
-                    startSeqNumber, curAppSeqNumber);
-            GetDeltaResponse confResponse = calculateConfigurationDelta(metaData, confSyncRequest, profile, historyDelta, curAppSeqNumber);
+    @Override
+    public SyncContext syncConfiguration(SyncContext context, ConfigurationClientSync request) throws GetDeltaException {
+        if (request != null) {
+            EndpointProfileDto profile = context.getEndpointProfile();
+            ClientSyncMetaData md = context.getMetaData();
+            AppSeqNumber appSeqNumber = cacheService.getAppSeqNumber(md.getApplicationToken());
+            if (context.getAppSeqNumber() == null) {
+                context.setAppSeqNumber(appSeqNumber);
+            }
+            int curAppSeqNumber = appSeqNumber.getSeqNumber();
+            LOG.debug("[{}][{}] fetched app seq number {}", context.getEndpointKey(), context.getRequestHash(), curAppSeqNumber);
+            LOG.trace("[{}][{}] procesing configuration sync request {}.", context.getEndpointKey(), context.getRequestHash(), request);
+            LOG.debug("[{}][{}] fetching history for seq numbers {}-{}", context.getEndpointKey(), context.getRequestHash(),
+                    request.getAppStateSeqNumber(), curAppSeqNumber);
+            int startSeqNumber = Math.min(request.getAppStateSeqNumber(), profile.getCfSequenceNumber());
+            LOG.debug("[{}][{}] calculating configuration delta using seq number {}", context.getEndpointKey(), context.getRequestHash(),
+                    startSeqNumber);
+            request.setAppStateSeqNumber(startSeqNumber);
```

Comment:
Yes, because profile and request seq numbers may be not synchronized. We need to take the smallest one to make sure we will not miss any updates.